projects/npm-tools/packages/npm-scripts/src/index.js

```diff
@@ -32,6 +32,10 @@ module.exports = async function () {
 			await require('./scripts/prettier')(...ARGS_ARRAY.slice(1));
 		},
 
+		async scanExposes() {
```

Comment:
Is there a more common name for this in our already existing Gradle plugins or tasks? `scanExposes` Something with `dependencies` or something like that?

Reply:
There's nothing like this anywhere (that I know of).

But we can still call it something more generic (like `scanDependencies` as you propose) in case we want to add more things in the future. For example, if we decide that this will not only scan needed `expose`s but also needed `remote`s, we can add that feature here.

Reply:
Yeah, I was thinking this is also kind of the things SF (or bndtools) does, so it could make sense that this would actually be integrated in the "check"/"format" scripts rather than being an independent tool.

Reply:
:thinking: Not sure about that, because the algorithm is too simple to assume that it will always be correct.

It will -for sure- miss some necessary cases (for example: dynamic strings containing module names, modules used in Java, etc.) but then, it can also show some false positives for any taglib that uses an attribute called `module` or `propsTransformer`.

It is possible that we may fix false positives by looking if the name is a valid `require`able module but we would need to check. As it is now, I consider it too fragile to be included in a SF.

But yeah, would be great to be able to have a SF covering as much use cases as possible.

Reply:
I'm thinking now that we may also define some kind of SF exclusion for cases where false positive shows up.

Reply:
As I suspected, this is not as easy as we may think. If we want to scan JSPs and fail the SF, we need to make sure that we correctly parse the taglibs and that's not an easy task because a tag invocation may have `<` and `>` signs inside the attribute values so we must parse attributes too. 

Think of this code:

```JSP
<my:taglib module="<%= myModule %>" message="Use < and > to move inside the UI">
     var module = "<>";
     ...
</my:taglib>
```

There are many places where a simple parser would fail but we need to make sure we don't fail in those cases.

So I will delay this task for a following PR because we need the other changes to continue with the epic.

We have [IFI-2327](https://issues.liferay.com/browse/IFI-2327) for this task, so I will revert it to open and remove the commit from this PR to be able to merge this.

Reply:
Also, for the record, @wincent suggested in a Slack conversation that this could also be done in the Java SF given that currently we only lint JS code inside the JSP files, not the files theirselves.

However, given that this will probably be needed for builds other than `liferay-portal` (GS, customers, etc.) I think we will need to move it here anyway.

Reply:
> this is not as easy as we may think

Indeed. We face the same problem with linting and formatting JS inside JSP — you can have arbitrary ~garbage~ JSP tags/expressions/scriptlets and EL expressions mixed in _anywhere_. Most of the complexity in the JS-in-JSP handling stems from dealing with that.